stdeb/util.py

```diff
@@ -489,7 +489,7 @@ def expand_zip(zip_fname,cwd=None):
     )
     contents = []
     for line in res.stdout.readlines()[3:-2]:
-        contents.append(line.split()[-1])
+        contents.append(b''.join(line.split()[3:]))
```

Comment:
Can you please explain why the index has been changed?

Reply:
Of course. Actually, the index doesn't really change.
The unzip -l command returns that (example taken from setuptools package v42.2.0):
`Archive:  setuptools-42.0.2.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
        0  2019-12-01 15:13   setuptools-42.0.2/
        0  2019-12-01 15:13   setuptools-42.0.2/docs/
        0  2019-12-01 15:13   setuptools-42.0.2/setuptools/
        0  2019-12-01 15:13   setuptools-42.0.2/pkg_resources/
        0  2019-12-01 15:13   setuptools-42.0.2/setuptools.egg-info/
[...]
     1466  2019-12-01 15:12   setuptools-42.0.2/docs/development.txt
     5639  2019-12-01 15:12   setuptools-42.0.2/docs/developer-guide.txt
      **218  2019-12-01 15:12   setuptools-42.0.2/setuptools/script (dev).tmpl**
     8184  2019-12-01 15:12   setuptools-42.0.2/setuptools/command/develop.py
[...]
     3206  2019-12-01 15:13   setuptools-42.0.2/setuptools.egg-info/entry_points.txt
---------                     -------
  2564657                     223 files
`
As you can see in the bold line, the the filename contains space, so in your version, it wiil take only the last element of the splited line : (dev).tmpl, but it should have taken script (dev).tmpl.

So that's why I take the third element and all that follows, that I join by '' to keep the full name of the file.

Reply:
Thanks for the explanation. Based on that context I think the patch is incorrect. After calling split and removing the first three items from the list the remaining once are: `['setuptools-42.0.2/setuptools/script', '(dev).tmpl']`. Joining them with an empty separate will effectively remove the space from the filename.

Instead I would suggest to invoke `split` with a maximum split of 3 (`.split(None, 3)` and then the resulting list will have exact one item containing the filename including spaces.

Have you tried to run `stdeb` with your patch and checked that it handles zip files correctly then? If not, please do so and comment what commands you used and the results you got.

Reply:
You're right. I have to join with a space instead of an empty character, my bad.
I've updated my pull request.
But I tried this patch to build setuptools package, and it work perfectly.
I'm not going to paste all the output (too verbose), but you can try to build setuptools debian package for example. Let's see some extract of my test.

With last stdeb release (0.9.0):
py2dsc-deb setuptools-42.0.2.zip 
```
Archive:  /tmp/setuptools-42.0.2.zip
   creating: /tmp/deb_dist/tmp_py2dsc/stdeb_tmp/setuptools-42.0.2/setuptools-42.0.2/
   creating: /tmp/deb_dist/tmp_py2dsc/stdeb_tmp/setuptools-42.0.2/setuptools-42.0.2/docs/
   creating: /tmp/deb_dist/tmp_py2dsc/stdeb_tmp/setuptools-42.0.2/setuptools-42.0.2/setuptools/
   creating: /tmp/deb_dist/tmp_py2dsc/stdeb_tmp/setuptools-42.0.2/setuptools-42.0.2/pkg_resources/
 [...]  
  inflating: /tmp/deb_dist/tmp_py2dsc/stdeb_tmp/setuptools-42.0.2/setuptools-42.0.2/setuptools.egg-info/SOURCES.txt  
  inflating: /tmp/deb_dist/tmp_py2dsc/stdeb_tmp/setuptools-42.0.2/setuptools-42.0.2/setuptools.egg-info/requires.txt  
  inflating: /tmp/deb_dist/tmp_py2dsc/stdeb_tmp/setuptools-42.0.2/setuptools-42.0.2/setuptools.egg-info/entry_points.txt  

running the following command in directory: deb_dist/tmp_py2dsc/setuptools-42.0.2
/home/jlejeune/.local/share/virtualenvs/build-deb/bin/python3 setup.py --command-packages stdeb.command sdist_dsc --dist-dir=/tmp/deb_dist --use-premade-distfile=/tmp/setuptools-42.0.2.zip bdist_deb

/home/jlejeune/.local/share/virtualenvs/build-deb/bin/python3: can't open file 'setup.py': [Errno 2] No such file or directory
ERROR running: /home/jlejeune/.local/share/virtualenvs/build-deb/bin/python3 setup.py --command-packages stdeb.command sdist_dsc --dist-dir=/tmp/deb_dist --use-premade-distfile=/tmp/setuptools-42.0.2.zip bdist_deb
ERROR in deb_dist/tmp_py2dsc/setuptools-42.0.2
```

With my patch:
py2dsc-deb setuptools-42.0.2.zip 
```
# Same beginning but finish with success
[...]
dpkg-deb: building package 'python3-setuptools' in '../python3-setuptools_42.0.2-1_all.deb'.
 dpkg-genbuildinfo --build=binary
 dpkg-genchanges --build=binary >../setuptools_42.0.2-1_amd64.changes
dpkg-genchanges: info: binary-only upload (no source code included)
 dpkg-source --after-build setuptools-42.0.2
dpkg-source: info: using options from setuptools-42.0.2/debian/source/options: --extend-diff-ignore=\.egg-info$
dpkg-buildpackage: info: binary-only upload (no source included)
```

Reply:
I updated the patch with my previous recommendation, see f316ac8a55cefedb24430e11f00cc98dac73ef22. This avoids having to split and rejoin the parts after the dropped prefix.

Could you please double check that it works for you too. Thanks.

Reply:
Your change is fine for me. It works also as expected. You can merge it.

Reply:
Your change is fine for me. It works also as expected. You can merge it.